app/concepts/todo/create.rb

```diff
@@ -11,40 +21,25 @@ class Todo::Create < Trailblazer::Operation
   #         associate `todo_list` with `user`
   #         associate `todo` with `todo_list`
   def process(params)
-    # make sure title is not blank
-    return invalid! if params[:todo][:title].blank?
-
-    # build the todo
-    todo = @model = Todo.new(params.require(:todo).permit(:title, :description))
-
-    # find or create the parent list for the todo
-    if params[:todo_list_id].blank?
-      todo_list = TodoList.find_or_create_by(name: "Default To-do List")
-
-      # Find or create the owner of the todo_list:
-      ##### If current_user_id is present, fetch that user
-      ##### otherwise, create a new one
-      current_user =
-        if params[:current_user] == nil
-          if params[:current_user_id] == nil
-            User.create!(fullname: "Guest")
-          else
-            User.find(params[:current_user_id])
-          end
-        else
-          params[:current_user]
-        end
-
-      # assign the todo_list to the user
-      todo_list.user = current_user
-      todo_list.save!
-    else # i.e. if params[:todo_list_id] is not nil
-      todo_list = TodoList.find(params[:todo_list_id])
+    # please try to debugger the output of validate(params[:todo]) here
+    # it should be false if there is validation error, true otherwise
+    validate(params[:todo]) do
+      contract.save
     end
-
-    # associate the todo with the todo_list
-    todo.list = todo_list
-    todo.save!
   end
 
-end
+  private
+
+    def setup_model!(todo_params)
+      # find or create the parent list for the todo
+      if todo_params[:todo_list_id].present?# i.e. if params[:todo_list_id] is not nil
+        todo_list = TodoList.find(todo_params[:todo_list_id])
+      elsif TodoList.find_by_name("Default To-do List").present?
+        todo_list = TodoList.find_by_name("Default To-do List")
+      else
+        result, operation = TodoList::Create.run(todo_list: { name: "Default To-do List" })
+        todo_list = operation.model
+      end
+      model.list = todo_list
+    end
+end
```

Comment:
Looks good. We prefer descriptive names over comment and we want to avoid comments if we can. Can you get rid of the comments. It's nice that you're able to use `setup_model!` Looks like you really dug deep. Currently, this class knows about when to create a todo_list. Can we move that logic to inside TodoList?